controller/resource.go

```diff
@@ -276,48 +261,43 @@ func (s *ServiceInfo) Delete() error {
 		s.Log.Info("token deleted successfully", "service", s.ServiceName)
 	}
 
+	return nil
+}
+
+// readSecret updates the AWS secret for the given service has a Token and
+// AccessorID if it is unset. If the secret is already set, this does nothing.
+func (s *ServiceInfo) readSecret() (tokenSecretJSON, error) {
+	var currSecret tokenSecretJSON
 	secretName := s.secretName()
-	s.Log.Info("updating secret", "name", secretName, "service", s.ServiceName)
-	_, err := s.SecretsManagerClient.UpdateSecret(&secretsmanager.UpdateSecretInput{
-		SecretId:     aws.String(secretName),
-		SecretString: aws.String(`{}`),
-	})
+
+	// Get current secret from AWS.
+	currSecretValue, err := s.SecretsManagerClient.GetSecretValue(&secretsmanager.GetSecretValueInput{SecretId: aws.String(secretName)})
 	if err != nil {
-		return fmt.Errorf("updating secret: %s", err)
+		return currSecret, fmt.Errorf("retrieving secret: %w", err)
 	}
-	s.Log.Info("secret updated successfully", "name", secretName, "service", s.ServiceName)
-
-	return nil
+	err = json.Unmarshal([]byte(*currSecretValue.SecretString), &currSecret)
+	if err != nil {
+		return currSecret, fmt.Errorf("unmarshalling secret value: %w", err)
+	}
+	return currSecret, nil
 }
 
-// updateServiceToken create a token in Consul and updates AWS secret with token's contents.
-func (s *ServiceInfo) updateServiceToken(secretName string) error {
+// updateServiceToken inserts an ACL token into Consul. The AccessorID and
+// SecretID are set based on the AWS secret.
+func (s *ServiceInfo) updateServiceToken(secret tokenSecretJSON) error {
```

Comment:
```suggestion
func (s *ServiceInfo) createServiceToken(secret tokenSecretJSON) error {
```

Since it is directly creating the token.